src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -718,10 +720,64 @@ InterpreterSelectQuery::InterpreterSelectQuery(
     sanitizeBlock(result_header, true);
 }
 
+void InterpreterSelectQuery::executePutInCache(QueryPlan & query_plan)
+{
+    auto settings = context->getSettingsRef();
+    QueryCachePtr query_cache = context->getQueryCache();
+    if (!settings.query_cache_active_usage || query_cache == nullptr)
+    {
+        return;
+    }
+
+    auto query_cache_key = CacheKey{query_ptr, result_header, settings,
```

Comment:
The MySQL query result cache (which they killed in the meantime) excluded queries with non-deterministic functions (https://dev.mysql.com/doc/refman/5.6/en/query-cache-operation.html). I think we should do that as well (and additionally I'd also put all encryption/decryption functions on the blocklist so that we never store cleartext in the cache - this could get funny with user sharing).